src/driver/queue.rs

```diff
@@ -0,0 +1,71 @@
+#[cfg(not(feature = "sync-queue"))]
+use std::{cell::UnsafeCell, collections::VecDeque};
+
+#[cfg(feature = "sync-queue")]
+use crossbeam_queue::SegQueue;
+
+#[cfg(feature = "sync-queue")]
+#[repr(transparent)]
+pub(super) struct Queue<T>(SegQueue<T>);
+
+#[cfg(feature = "sync-queue")]
+impl<T> Queue<T> {
+    pub fn with_capacity(_capacity: usize) -> Self {
+        Self(SegQueue::new())
+    }
+
+    pub fn push(&self, value: T) {
+        self.0.push(value);
+    }
+
+    pub fn pop(&self) -> Option<T> {
+        self.0.pop()
+    }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+}
+
+#[cfg(not(feature = "sync-queue"))]
+#[repr(transparent)]
+pub(super) struct Queue<T>(UnsafeCell<VecDeque<T>>);
```

Comment:
Well, will it be safer if we use `RefCell`? `UnsafeCell` *should* be OK but I'm not that confident.

Reply:
It will be safe when single thread invariant is violated but will not be zero cost - RefCell maintains a borrow flag.

The expectation is that the runtime is single threaded and we don't use signal feature. If this holds Queue is not accessed concurrently.

Reply:
OK, fair enough.